fxa/oauth.py

```diff
@@ -148,10 +148,16 @@ def authorize_code(self, sessionOrAssertion, scope=None, client_id=None,
             client_id = self.client_id
         assertion = self._get_identity_assertion(sessionOrAssertion, client_id)
         url = "/authorization"
+
+        # Although not relevant in this scenario from a security perspective,
+        # we generate a random 'state' and check the returned redirect URL
+        # for completeness.
+        state = base64.urlsafe_b64encode(os.urandom(23)).decode('utf-8').rstrip("=")
```

Comment:
If you use a number of bytes that is a multiple of 6 you won't have base64 padding.

```suggestion
        state = base64.urlsafe_b64encode(os.urandom(24)).decode('utf-8')
```